### ISSUE TITLE
Improve followers feature : consider more interactions  

### ISSUE THREAD

**Issue by lapineige** [kensanata/mastodon-archive#97]
Hello,

I really like the followers feature https://github.com/kensanata/mastodon-archive#followers as a semi-automated cleanup tool.
It adds some nice improvement over Mastodon integrated feature (that did not exist at the time you created this) as it allows to filter interactions older than a given time (only 1 month in Mastodon).

I would like to suggest some improvements:
- as of Mastodon, the ability to filter followers-only, or mutual.
- as of Mastodon, the ability not to block, but to remove the follow (with or without removing the following). As an user, in many cases I don't want to block the person, so they can come back afterwards and ask for a follow again
- include favourites and possibly boosts in the interactions. It help both to detect lurking bots/accounts (which tends not to interact at all) and to allow people who are following "just to listen"/to be able to read private conversation mentioning you.
- include mentions, not only the incoming ones as of right now, but also when we mention them. This also helps distinguishing between real accounts and the rest.
- find a way to exclude secondary accounts and migrated ones ? Those obviously won't interact often, but you might want to keep them.
I don't even know if we can easily find the migrated ones with or without an API request.
As a minimum, maybe we could exclude any same-pseudo matching account, that would reduce the number of false positives.

*I'm documenting this here, as I might work on it in the future but don't have time now, just in case someone would like to help… :slightly_smiling_face:*

Let me know what you think.

Have a nice day :)

**Comment by kensanata** [comment]
It sounds interesting, for sure! Specially if this is how we can get advanced features to other servers, like GoToSocial.

**Comment by lapineige** [comment]
> Specially if this is how we can get advanced features to other servers, like GoToSocial.

I don't really get your point.

Does that mean you want/plan to be able to archive other fediverse software content, or *any* Mastodon-API compatible software ?

**Comment by kensanata** [comment]
Yes, as long as the server supports the Mastodon Client API, mastodon-archive should be able to handle it. And I personally use a single person GoToSocial. But in that case the copy of the database is already in my hands so perhaps I’ll write a different set of tools instead. 😅

**Comment by lapineige** [comment]
> Yes, as long as the server supports the Mastodon Client API, mastodon-archive should be able to handle it.

That's great !!
